src/main/java/com/sap/oss/phosphor/fosstars/data/artifact/ReleaseInfoFromMaven.java

```diff
@@ -61,10 +64,34 @@ private JsonNode mavenArtifactReleaseInfo(MavenArtifact mavenArtifact) throws IO
         }
       }
       values.update(RELEASED_ARTIFACT_VERSIONS.value(new ArtifactVersions(artifactVersions)));
+      updateArtifactVersion(mavenArtifact, artifactVersions, values);
       return this;
     }
 
     values.update(RELEASED_ARTIFACT_VERSIONS.unknown());
+    values.update(ARTIFACT_VERSION.unknown());
     return this;
   }
+
+  /**
+   * Update the {@link com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures#ARTIFACT_VERSION}
+   * value based on given parameters.
+   * If version of MavenArtifact is not in the set of artifact versions or the MavenArtifact
+   * does not have a version set the ARTIFACT_VERSION is set to unknown.
+   *
+   * @param mavenArtifact The maven artifact.
+   * @param artifactVersions All found artifact versions.
+   * @param values The set of values to be updated.
+   */
+  private static void updateArtifactVersion(MavenArtifact mavenArtifact,
```

Comment:
`ReleaseInfoFromNpm` has the same method but it takes `NmpArtifact`. Could you please move this method to `AbstractReleaseInfoLoader`? That would remove duplicate code. The method cat take `Artifact` or a version string.

Reply:
I thought about adding a `version()` method to the `Artifact` interface.
But IMHO it looks strange to have a version but no identifier.
And as we have not decided if a purl may be an option (as general identifier) I have no replaced the `(Maven)Artifact` with the `Optional<String> version`.
Would like to keep the optional as then the code in the specific `AbstractReleaseInfoLoader` implementation can be kept simpler.
Just let me know if this is fine for you.

Reply:
Yes, that's okay.